test/Token.t.sol

```diff
@@ -440,4 +440,97 @@ contract TokenTest is NounsBuilderTest, TokenTypesV1 {
         vm.prank(otherUsers[0]);
         auction.unpause();
     }
+
+    function testRevert_OnlyOwnerUpdateFounders() public {
+        deployMock();
+
+        address f1Wallet = address(0x1);
+        address f2Wallet = address(0x2);
+        address f3Wallet = address(0x3);
+
+        address[] memory founders = new address[](3);
+        uint256[] memory percents = new uint256[](3);
+        uint256[] memory vestingEnds = new uint256[](3);
+
+        founders[0] = f1Wallet;
+        founders[1] = f2Wallet;
+        founders[2] = f3Wallet;
+
+        percents[0] = 1;
+        percents[1] = 2;
+        percents[2] = 3;
+
+        vestingEnds[0] = 4 weeks;
+        vestingEnds[1] = 4 weeks;
+        vestingEnds[2] = 4 weeks;
+
+        setFounderParams(founders, percents, vestingEnds);
+
+        vm.prank(f1Wallet);
+        vm.expectRevert(abi.encodeWithSignature("ONLY_OWNER()"));
+
+        token.updateFounders(foundersArr);
+    }
+
+    function test_UpdateFounderShareAllocationFuzz(
+        uint256 f1Percentage,
+        uint256 f2Percentage,
+        uint256 f3Percentage
+    ) public {
+        deployMock();
+
+        address f1Wallet = address(0x1);
+        address f2Wallet = address(0x2);
+        address f3Wallet = address(0x3);
+
+        vm.assume(f1Percentage > 0 && f1Percentage < 100);
+        vm.assume(f2Percentage > 0 && f2Percentage < 100);
+        vm.assume(f3Percentage > 0 && f3Percentage < 100);
```

Comment:
If you pass in 1 and 2 founder ownership percentage the third one can be implicitly found through 100 - f1 - f2 = f3

Reply:
This assumes we want `sum(f1, f2, f3) = 100` though, no? Would break the test cases for [1, 1, 1], for example

Reply:
My bad — carry on—